src/time.cpp

```diff
@@ -32,9 +32,33 @@ POSSIBILITY OF SUCH DAMAGE.
 
 #include "libtorrent/aux_/time.hpp"
 
+#include <chrono>
+
 namespace libtorrent { namespace aux {
 
 	time_point time_now() { return clock_type::now(); }
 	time_point32 time_now32() { return time_point_cast<seconds32>(clock_type::now()); }
 
+	// for simplying implementation
+	using std::chrono::system_clock;
+
+	time_t to_time_t(const time_point32 tp)
+	{
+		const auto hc_now = clock_type::now();
+		const auto sys_now = system_clock::now();
+
+		const auto r = sys_now + std::chrono::duration_cast<system_clock::duration>(hc_now - tp);
+		return system_clock::to_time_t(r);
+	}
+
+	time_point32 from_time_t(const std::time_t t)
+	{
+		const auto tp = system_clock::from_time_t(t);
+		const auto hc_now = std::chrono::high_resolution_clock::now();
```

Comment:
I believe this should use `lt::clock_type::now()`, right? To preserve the abstraction of `lt::clock_type`. It may be changed to a `monotonic_clock` in the future for instance